subprojects/test-runner/test-inhouse-runner/src/main/kotlin/com/avito/android/rule/ActivityScenarioRule.kt

```diff
@@ -0,0 +1,23 @@
+package com.avito.android.rule
+
+import android.app.Activity
+import androidx.test.espresso.intent.Intents
+import com.avito.android.rule.base.BaseActivityScenarioRule
+
+class ActivityScenarioRule<T : Activity>(activityClass: Class<T>) : BaseActivityScenarioRule<T>(
+    activityClass = activityClass,
+    initialTouchMode = true,
+    launchActivity = false,
+) {
+
+    override fun afterActivityLaunched() {
+        Intents.init()
```

Comment:
What does that code do?

Reply:
```
/**
   * Initializes Intents and begins recording intents. Must be called prior to triggering any
   * actions that send out intents which need to be verified or stubbed. This is similar to
   * MockitoAnnotations.initMocks.
   */
   ```

Reply:
Why do we need that code?

Reply:
As I see, it just simplifies test writing with minimal performance penalties. Test authors don't copy this boilerplate code and have no chance to make mistakes, such as forgetting to call the `release` function.

Reply:
We don't. There is an issue(bug) about this,
but i believe it should be addressed in separate PR